src/SplineMap.cc

```diff
@@ -0,0 +1,60 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+#include <sstream>
+#include <stdexcept>
+
+#include "astshim/SplineMap.h"
+
+namespace ast {
+
+SplineMap::SplineMap(AstSplineMap *map) : Mapping(reinterpret_cast<AstMapping *>(map)) {
+    if (!astIsASplineMap(getRawPtr())) {
+        std::ostringstream os;
+        os << "this is a " << getClassName() << ", which is not a SplineMap";
+        throw std::invalid_argument(os.str());
+    }
+}
+
+/// Make a raw AstSplineMap with a specified forward transform.
+AstSplineMap *SplineMap::_makeRawSplineMap(int kx, int ky, int nx, int ny, std::vector<double> const &tx,
+    std::vector<double> const &ty, std::vector<double> const &cu, std::vector<double> const &cv,
+    std::string const &options) const {
+
+    const int nTx = tx.size();
```

Comment:
These do a narrowing cast.
`size()` returns `size_t` (unsigned 64 bit)
`int` is 32 bit (signed)
Likely OK as the vectors are not big.